client/src/codegen/go-client.ts

```diff
@@ -705,16 +723,23 @@ type Client struct {
 }
 
 type Options struct {
-	Endpoint string
+  Endpoint string
+  Secret   string
 }
 
 func New(options *Options, opts ...graphql.ClientOption) *Client {
-	endpoint := DefaultEndpoint
+  endpoint := DefaultEndpoint
+  secret   := Secret
 	if options != nil {
-		endpoint = options.Endpoint
+    if options.Endpoint != "" {
```

Comment:
Has what I discussed with @schickling changed? We had agreed that if the user provided a non-nil `options`, that all values would have to be provided, and not have defaults.

Reply:
Indeed, you are correct, I will make amends.